packages/snaps-utils/src/fs.test.ts

```diff
@@ -207,3 +206,37 @@ describe('validateDirPath', () => {
     );
   });
 });
+
+describe('useTemporaryFile', () => {
+  it('creates a temporary file with the proper name and content', async () => {
+    await useTemporaryFile('foo', 'bar', async (filePath) => {
+      expect(filePath).toContain('foo');
+      const content = await fs.readFile(filePath, { encoding: 'utf-8' });
+      expect(content.toString()).toBe('bar');
+    });
+    expect.assertions(2);
+  });
+
+  it('always deletes the temporary file after usage', async () => {
+    let filePath: string;
+
+    await useTemporaryFile('foo', 'bar', async (fp) => {
+      filePath = fp;
```

Comment:
Just to make sure the callback has been called, can we add an assertion here, and increase the number of expected assertions?

```suggestion
      expect(fp).toBeDefined();
      filePath = fp;
```